### PR TITLE
[4.0] php-cs-fixer v3.0 Support

### PR DESCRIPTION
**Pull request by 810**
Pull Request for Issue # .

### Summary of Changes

Now supporting new php-cs-fixer version.

### Testing Instructions

Joomla install and then setup you local project.
composer install - then do composer update

### Actual result BEFORE applying this Pull Request

depricated PhpCsFixer\Config

### Expected result AFTER applying this Pull Request

No errors.

### CODE REVIEW

**Comment by PhilETaylor** [comment]
well if this is actually going to be done, then please check https://github.com/joomla/joomla-cms/pull/33362/files and merge the good parts without the controversy ... the fact is that this project doesn't use php-cs-fixer at all and there is [no reason why this configuration should be in this repo](https://github.com/joomla/joomla-cms/pull/33374), its not used, not even by the maintainers of the CI.

File: .php-cs-fixer.dist.php [hunk 1]
```diff
@@ -0,0 +1,116 @@
+<?php
+/**
+ * @package    Joomla.Site
+ *
+ * @copyright  (C) 2016 Open Source Matters, Inc. <https://www.joomla.org>
```
Comment (brianteeman) [inline .php-cs-fixer.dist.php]:
```suggestion
 * @copyright  (C) 2021 Open Source Matters, Inc. <https://www.joomla.org>
```

**Comment by brianteeman** [comment]
guessing that `.php-cs-fixer.dist.php` should be added the list of files not to be included in a release

**Comment by brianteeman** [comment]
I have tested this item :white_check_mark: successfully on 5734a37762f209f781038193671d0f65d098f324<br /><br />Thank you<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/33908">issues.joomla.org/tracker/joomla-cms/33908</a>.</sub>

**Comment by brianteeman** [comment]
NOTE: I only tested that the PR worked as described. I do not know if the rules are correct for Joomla.

File: .php-cs-fixer.dist.php [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+ *
+ *        ./libraries/vendor/bin/php-cs-fixer fix
+ *
+ * You can run the clean up on a single file if you need to, this is faster
+ *
+ *        ./libraries/vendor/bin/php-cs-fixer fix --dry-run administrator/index.php
+ *        ./libraries/vendor/bin/php-cs-fixer fix administrator/index.php
+ */
+
+// Only index the files in /libraries and no deeper, to prevent /libraries/vendor being indexed
```
Comment (brianteeman) [inline .php-cs-fixer.dist.php:32]: dont we also want to include /libraries/src

Comment (brianteeman) [reply .php-cs-fixer.dist.php:32]: oop

**Comment by brianteeman** [comment]
i wish this could be merged and we can fix all the issues

**Comment by 810** [comment]
Please merge this.

**Comment by wilsonge** [comment]
Thanks!

**Pull request merged**